Python/gc_free_threading.c

```diff
@@ -923,6 +928,41 @@ gc_should_collect(GCState *gcstate)
             gcstate->generations[1].threshold == 0);
 }
 
+static void
+record_allocation(PyThreadState *tstate)
+{
+    struct _gc_thread_state *gc = &((_PyThreadStateImpl *)tstate)->gc;
+
+    // We buffer the allocation count to avoid the overhead of atomic
+    // operations for every allocation.
+    gc->alloc_count++;
+    if (gc->alloc_count >= LOCAL_ALLOC_COUNT_THRESHOLD) {
+        // TODO: Use Py_ssize_t for the generation count.
+        GCState *gcstate = &tstate->interp->gc;
+        _Py_atomic_add_int(&gcstate->generations[0].count, (int)gc->alloc_count);
```

Comment:
Is there any overflow threshold we need to worry about here on `gcstate->generations[0].count`?

Reply:
Yeah, that's why I left a TODO.

I'm trying to avoid substantial changes to `pycore_gc.h` until Mark lands his incremental GC in order to avoid merge conflicts.